doc/src/explanation/glossary.md

```diff
@@ -0,0 +1,712 @@
+    expression. There are different methods in SymPy for performing
+    substitution, including {meth}`subs <sympy.core.basic.Basic.subs>`,
+    {meth}`replace <sympy.core.basic.Basic.replace>`, and {meth}`xreplace
+    <sympy.core.basic.Basic.xreplace>`. The methods may differ depending on
+    whether they perform substitution using only strict {term}`structural
+    equality` or by making use of mathematical knowledge when determining
+    where a subexpression appears in an expression. Substitution is the
+    standard way to treat an expression as a mathematical {term}`function` and
+    evaluate it at a point.
+
+Symbolic
+
+    A *symbolic* representation of a mathematical object is a representation
+    that is partially or completely unevaluated at runtime. It may include
+    named {term}`symbolic constants <symbol>` in place of explicit numeric
+    values. A symbolic representation is often contrasted with a
+    {term}`numeric` one. Symbolic representations are mathematically exact, to
+    contrast with numeric representations which are typically rounded so they
+    can fit within a floating-point value. Symbolic {term}`expressions
+    <expression>` representing mathematical objects may be aware of
+    mathematical properties of these objects and be able to {term}`simplify
+    <simplification>` to equivalent symbolic expressions using those
+    properties. The goal of SymPy is to represent and manipulate symbolic
+    expressions representing various mathematical objects.
+
+    Some sources use the phrases "analytic solution" or "closed-form" to refer
+    to the concept of "symbolic", but this terminology is not used in SymPy.
+    If used in SymPy, "analytic" would refer to the property of being [an
+    analytic function](https://en.wikipedia.org/wiki/Analytic_function), and
+    in SymPy {term}`solve` refers only to a certain type of symbolic
+    operation. "Closed-form" in SymPy would typically refer the mathematical
+    sense of the term, whereas "symbolic" would generally refer to the
+    implementation detail of how a mathematical concept is implemented, and be
+    in contrast with a {term}`numeric` implementation of the same mathematical
+    concept.
+
+{class}`~.Symbol`
+
+    *{class}`~.Symbol`* is the class for symbol objects. A symbol represents a
+    single mathematical variable in an expression. The {class}`~.Symbol` class
+    is a subclass of {term}`Expr` and is {term}`atomic <atom>`. A `Symbol`
+    contains a name, which is any string, and {term}`assumptions`. Symbols are
+    typically defined with the `Symbol` constructor or the {func}`~.symbols`
+    function. Two Symbols with the same name and assumptions are considered
+    {term}`equal <structural equality>`. Symbols are implicitly assumed to be
+    independent or constant with respect to one another. SymPy does not
+    distinguish between symbols that are constants, variables, or parameters.
```

Comment:
In the core no such distinction is made but when calling a function like solve these are distinguished.

Reply:
Maybe I should word this better. They're not intrinsically distinguished. A symbol being constant or not depends on how it is used (e.g., if it is passed as an argument to solve or not). It's a common misconception I see from new users that they need to specify that a symbol is constant or not when defining it.